shopinvader_assortment/models/se_backend.py

```diff
@@ -0,0 +1,67 @@
+# Copyright 2018 ACSONE SA/NV
+# Copyright 2021 Camptocamp (http://www.camptocamp.com).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+from odoo.osv import expression
+
+
+class SeBackend(models.Model):
+    _inherit = "se.backend"
+
+    product_manual_binding = fields.Boolean(default=True)
+    product_assortment_id = fields.Many2one(
+        string="Product Assortment",
+        comodel_name="ir.filters",
+        help="Bind only products matching with the assortment domain",
+        domain=[("is_assortment", "=", True)],
+        context={"product_assortment": True},
+    )
+
+    @api.constrains("product_manual_binding", "product_assortment_id")
+    def _check_product_assortment_id(self):
+        if not all(
+            rec.product_manual_binding or rec.product_assortment_id for rec in self
+        ):
+            raise ValidationError(
+                _("Product Assortment is required for automatic binding")
+            )
+
+    def _autobind_product_from_assortment(self, domain_product=None):
+        self.ensure_one()
+        if not self.product_assortment_id:
+            return
+        if domain_product is None:
+            domain_product = []
+        domain_assortment = self.product_assortment_id._get_eval_domain()
+        domain = expression.AND([domain_assortment, domain_product])
+        product_obj = self.env["product.product"]
+        assortment_products = product_obj.search(domain)
+        product_indexes = self.index_ids.filtered(
+            lambda i: i.model_id.model == "product.product"
+        )
+        products_bound = (
+            product_indexes.binding_ids.record
+            if product_indexes.binding_ids
+            else product_obj
+        )
```

Comment:
@sbejaoui  The use of the ORM for such a logic will introduce performance costs

Reply:
@sbejaoui you can find [here] (https://github.com/OCA/product-attribute/pull/1352/files#diff-71273ac18558d72e9ced9f26e9d84468a90f9bfd25d53c4ba9d3b85ff4540252R174) a way to write performant queries to get products to bind and to unbind....